### DISCUSSION TITLE
New release v1.2 ‼️

### DISCUSSION THREAD

**Discussion by jczic** [jczic/ESP32-MPY-Jama#40] [Announcements]
# What's new in version 1.2.0

## Significant changes:
- UI enhanced (colors, icons, animations, etc)
- Several performance improvements to the communication with the ESP32
- Management of the automatic configuration of the components at startup (Wi-Fi, AP, Ethernet, SD card)
- Much more robust REPL terminal display management
- Support for Ethernet PHY interfaces
- Real time on networks panel and new information displayed (RSSI, AP clients, etc)
- New page to access the latest releases and useful links of the eco-system
- Optimizations for REPL terminal UI to avoid text flooding
- Displaying information about flash memory partitions
- Fast automatic reconnection system to ESP32 boards
- Many memo images of GPIO pinouts of boards
- SD card file system support (init, mount, release, format)
- Much more reliable and robust serial port connection on many boards
- Improved memory management on ESP32
- New execution method for JAMA functions and programs on the device
- Instructions for Linux updated and build a binary added
- Better support on Linux
- Many bugs fixed!

## New JAMA Funcs:
- DAC Output (voltage)
- 1-Wire Devices Scan (IDs & SN)
- GPIO Input (high/low)
- ADC Reader (level & voltage)
- Wi-Fi Scan (detailled)
- KT403A MP3 Player (DFPlayer, Grove-MP3)
- LEDs - NeoPixel Strip (RGB+W/Y)
- LEDs - DotStar Strip (RGB)
- GPIO Output (high/low)
- BLE iBeacon v2 (name & Apple beacon)
- BLE Scan (a little enhanced)

## Other changes:
- Keys to escape, validate and close dialogue boxes
- Better management of the tab key in the IDE
- Argument for forcing the web engine GUI
- Improvement of the GPIO list in the system page
- Correct processing of a .py file directly executed from folders
- Better refresh of the folder/file list
- Device and MCU names displayed in the top bar
- Fixed Internet access verification problem on some firmwares
- Improvement of some Jama Funcs
- ...

**Comment by Avaray** [comment]
I just discovered this project. I really like your IDE. Nice work!

**Comment by jczic** [comment]
Merci beaucoup :)

Le mer. 17 avr. 2024 à 06:21, Avaray ***@***.***> a écrit :

> I just discovered this project. I really like your IDE. Nice work!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jczic/ESP32-MPY-Jama/discussions/40#discussioncomment-9137333>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAD2ALF4NZS7ECZ37XZLYDLY5X2FDAVCNFSM6AAAAAAW3UBE5OVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCMZXGMZTG>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*   Sincèrement,*
      Jean-Christophe Bos
      06 80 27 93 84

**Comment by TapioJarnfors** [comment]
Greetings from Finland! Nice work - this is a really useful "Swiss Army Knife" :)
